local/rest_api_gcbm/gcbm.py

```diff
@@ -371,50 +453,114 @@ def safe_read_json(file_path):
+        file_path="disturbances_2012.tiff",
+        category="disturbance",
+        payload={"year": 2012, "disturbance_type": "Wildfire", "transition": 1},
+    )
+
+    sim.add_file("tests/reference/test_data/new_demo_run/disturbances_2013.tiff")
+    # sim.set_attributes(
+    #     file_path="disturbances_2013.tiff",
+    #     category="disturbance",
+    #     payload={
+    #         "year": 2013,
+    #         "disturbance_type": "Mountain pine beetle — Very severe impact",
+    #         "transition": 1,
+    #     },
+    # )
 
     # TODO: Check how to handle multiple attributes entries (L442-451 of `app.py:master`)
-    # sim.set_disturbance_attributes("disturbances_2013_moja.tiff",
-    #                                 {"year": 2013, "disturbance_type": "Wildfire", "transition": 1})
-
-    sim.add_file("tests/tiff/new_demo_run/disturbances_2014_moja.tiff")
-    sim.set_disturbance_attributes("disturbances_2014_moja.tiff",
-                                    {"year": 2014, "disturbance_type": "Mountain pine beetle — Very severe impact", "transition": 1})
-
-    sim.add_file("tests/tiff/new_demo_run/disturbances_2015_moja.tiff")
-    sim.set_disturbance_attributes("disturbances_2015_moja.tiff",
-                                    {"year": 2015, "disturbance_type": "Wildfire", "transition": 1})
-
-    sim.add_file("tests/tiff/new_demo_run/disturbances_2016_moja.tiff")
-    sim.set_disturbance_attributes("disturbances_2016_moja.tiff",
-                                   {"year": 2016, "disturbance_type": "Wildfire", "transition": 1})
-
-    sim.add_file("tests/tiff/new_demo_run/disturbances_2018_moja.tiff")
-    sim.set_disturbance_attributes("disturbances_2018_moja.tiff",
-                                    {"year": 2018, "disturbance_type": "Wildfire", "transition": 1})
-
-    # TODO: classifiers don't have 'year' attributes
-    sim.add_file("tests/tiff/new_demo_run/Classifier1_moja.tiff", category="classifiers")
-    # sim.set_classifier_attributes("Classifier1_moja.tiff",
-    #                               {"1": "TA", "2": "BP", "3": "BS", "4": "JP", "5": "WS", "6": "WB", "7": "BF", "8": "GA"})
-
-    sim.add_file("tests/tiff/new_demo_run/Classifier2_moja.tiff", category="classifiers")
-    # sim.set_classifier_attributes("Classifier2_moja.tiff",
-    #                              {"1": "5", "2": "6", "3": "7", "4": "8"})
-
-    sim.add_file("tests/tiff/new_demo_run/initial_age_moja.tiff", category="miscellaneous")
-    sim.add_file("tests/tiff/new_demo_run/mean_annual_temperature_moja.tiff", category="miscellaneous")
+    sim.set_attributes(
+        file_path="disturbances_2013.tiff",
+        category="disturbance",
+        payload={
+            "1": {
+                "year": 2013,
+                "disturbance_type": "Mountain pine beetle — Very severe impact",
+                "transition": 1,
+            },
+            "2": {"year": 2013, "disturbance_type": "Wildfire", "transition": 1},
+        },
+    )
```

Comment:
Passing attributes in nested dictionary is now supported. It also checks if `year`/`{category}_type` are anywhere in the nested dictionary, and sets `has_year`/`has_type` keys in the config file respectively.